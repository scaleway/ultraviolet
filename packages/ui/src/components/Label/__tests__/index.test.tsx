import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithTheme } from '@utils/test'
import { describe, expect, test } from 'vitest'
import { Label } from '..'

describe('DateInput', () => {
  test('renders correctly with default props', () => {
    const { asFragment } = renderWithTheme(<Label>Label</Label>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly required', () => {
    const { asFragment } = renderWithTheme(<Label required>Label</Label>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with label description (string)', () => {
    const { asFragment } = renderWithTheme(
      <Label labelDescription="test">Label</Label>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with label description (ReactNode)', () => {
    const { asFragment } = renderWithTheme(
      <Label labelDescription={<div>test</div>}>Label</Label>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
  test('renders correctly small', () => {
    const { asFragment } = renderWithTheme(<Label size="small">Label</Label>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly medium', () => {
    const { asFragment } = renderWithTheme(<Label size="medium">Label</Label>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly large', () => {
    const { asFragment } = renderWithTheme(<Label size="large">Label</Label>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly with htmlFor', async () => {
    const { asFragment } = renderWithTheme(
      <div>
        <Label size="small" htmlFor="id" id="id-label">
          Label
        </Label>
        <input id="id" data-testid="test-input" />
      </div>,
    )

    const input = screen.getByTestId('test-input')
    const label = screen.getByText('Label')

    await userEvent.click(label)
    expect(input).toHaveFocus()
    expect(asFragment()).toMatchSnapshot()
  })
})
