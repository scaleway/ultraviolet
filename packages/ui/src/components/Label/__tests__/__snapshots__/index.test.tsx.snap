// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DateInput > renders correctly large 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <label
      class="emotion-0 emotion-1"
    >
      Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly medium 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #222638;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <label
      class="emotion-0 emotion-1"
    >
      Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly required 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  color: #b3144d;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <label
        class="emotion-2 emotion-3"
      >
        Label
      </label>
      <label
        class="emotion-4 emotion-3"
      >
        *
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly small 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #222638;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <label
      class="emotion-0 emotion-1"
    >
      Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <label
      class="emotion-0 emotion-1"
    >
      Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly with htmlFor 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #222638;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <div>
      <label
        class="emotion-0 emotion-1"
        for="id"
        id="id-label"
      >
        Label
      </label>
      <input
        data-testid="test-input"
        id="id"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly with label description (ReactNode) 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <label
        class="emotion-2 emotion-3"
      >
        Label
      </label>
      <div>
        test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DateInput > renders correctly with label description (string) 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.25rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <label
        class="emotion-2 emotion-3"
      >
        Label
      </label>
      <label
        class="emotion-4 emotion-3"
      >
        test
      </label>
    </div>
  </div>
</DocumentFragment>
`;
